internal/manager/utils/kongconfig/root.go

```diff
@@ -103,13 +110,16 @@ func RouterFlavorFromRoot(r Root) (string, error) {
 	const routerFlavorKey = "router_flavor"
 	routerFlavor, exist := rootConfig[routerFlavorKey]
 	if !exist {
-		return "", fmt.Errorf("no value in root configuration for key %q", routerFlavorKey)
+		if kongVersion.Major() < expressionRouterMinimalMajorVersion {
+			return kongRouterFlavorTraditional, nil
+		}
+		return kongRouterFlavorTraditionalCompatible, nil
```

Comment:
As discussed on slack: since Gateway 3.0 and above should have this field, I believe this is a good place to return an error. Otherwise, return `kongRouterFlavorTraditional`

```suggestion
		if kongVersion.Major() < expressionRouterMinimalMajorVersion {
			return kongRouterFlavorTraditional, nil
		}
		return "", fmt.Errorf("%q field missing from Gateway's configuration root for version: %s", routerFlavorKey, kongVersion)
```

Reply:
If we do the above then changelog would also need to be updated.

Reply:
updated, and also for CHANGELOG.